### ISSUE TITLE
lucy-console: "SyntaxError: [object Object] is not a valid Swagger API definition"

### ISSUE THREAD

**Issue by nagromc** [lucybot/api-spec-converter#85]
Hi,

I try to use `api-spec-converter` (v. 1.8.0) with [`lucy-console`](/lucybot/lucy-console) (v. 1.1.1) but I run into the following error (in the web browser console when I reach [http://localhost:3000](http://localhost:3000)):

> SyntaxError: [object Object] is not a valid Swagger API definition

> 10:32:40,576 SyntaxError: [object Object] is not a valid Swagger API definition
> create/<() %5Cminified%5Cjs%5Cconsole.js:1909
> [1]</SwaggerParser.prototype.parse/<() %5Cminified%5Cjs%5Cconsole.js:1633
> 1 %5Cminified%5Cjs%5Cconsole.js:1909:318

I don't know if the problem comes from `api-spec-converter` that doesn't output Swagger-compliant object, or from `lucy-console` that cannot handle this particular Swagger specification. The thing is `lucy-console` works great with the example given in their [README.md](https://github.com/lucybot/lucy-console/blob/master/README.md#sample-usage).

My Gulp task:
```node
var ConsoleApp = require('express')();
var LucyConsole = require('lucy-console');
var Converter = require('api-spec-converter');

gulp.task('view-api', function() {
    Converter.convert({
        from: 'wadl',
        to: 'swagger_2',
        source: 'http://localhost:9080/rest/application.wadl' // returns [0]
    }, function(err, converted) {
        console.log(converted.stringify()); // outputs [1]

        var apiConsole = new LucyConsole({
            swagger: converted.spec
        });

        ConsoleApp.use(apiConsole.router);
        ConsoleApp.listen(3000);
    });
});
```

[0]
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<application xmlns="http://wadl.dev.java.net/2009/02">
    <doc xmlns:jersey="http://jersey.java.net/" jersey:generatedBy="Jersey: 1.19 02/11/2015 05:39 AM"/>
    <grammars>
        <include href="application.wadl/xsd0.xsd">
            <doc xml:lang="en" title="Generated"/>
        </include>
    </grammars>
    <resources base="http://localhost:9080/rest/">
        <resource path="/Affectation">
            <resource path="/articles/{identifiantUm}/{siteUM}">
                <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:string" style="template" name="identifiantUm"/>
                <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:string" style="template" name="siteUM"/>
                <method name="GET" id="rechercherArticle">
                    <response>
                        <ns2:representation xmlns:ns2="http://wadl.dev.java.net/2009/02" xmlns="" mediaType="application/json" element="bandeauRMVO"/>
                    </response>
                </method>
            </resource>
            <resource path="/articles">
                <method name="POST" id="rechercherArticle">
                    <request>
                        <ns2:representation xmlns:ns2="http://wadl.dev.java.net/2009/02" xmlns="" mediaType="application/json" element="resultRechercherArticlePB"/>
                    </request>
                    <response>
                        <ns2:representation xmlns:ns2="http://wadl.dev.java.net/2009/02" xmlns="" mediaType="application/json" element="resultatRechercherArticleVO"/>
                    </response>
                </method>
            </resource>
        </resource>
        <resource path="/rmes">
            <method name="GET" id="getRessourcesMetal">
                <response>
                    <representation mediaType="application/json"/>
                </response>
            </method>
            <resource path="/{identifiantUm}/{siteUm}">
                <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:string" style="template" name="identifiantUm"/>
                <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:string" style="template" name="siteUm"/>
                <method name="GET" id="get">
                    <response>
                        <representation mediaType="application/json"/>
                    </response>
                </method>
                <resource path="/categorisations">
                    <method name="GET" id="get">
                        <response>
                            <representation mediaType="application/json"/>
                        </response>
                    </method>
                </resource>
            </resource>
        </resource>
        <resource path="/demandesAffectation">
            <resource path="/{reference}">
                <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:int" style="template" name="reference"/>
                <method name="GET" id="get">
                    <response>
                        <representation mediaType="application/json"/>
                    </response>
                </method>
            </resource>
        </resource>
        <resource path="/apgs">
            <resource path="/{numApg: [\p{Alnum}]{1,20}}">
                <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:string" style="template" name="numApg"/>
                <method name="GET" id="getApg">
                    <response>
                        <representation mediaType="application/json"/>
                    </response>
                </method>
                <resource path="/rmes">
                    <method name="GET" id="getRessourcesMetal">
                        <response>
                            <representation mediaType="application/json"/>
                        </response>
                    </method>
                    <resource path="/{identifiantUm}/{siteUm}">
                        <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:string" style="template" name="identifiantUm"/>
                        <param xmlns:xs="http://www.w3.org/2001/XMLSchema" type="xs:string" style="template" name="siteUm"/>
                        <method name="GET" id="get">
                            <response>
<representation mediaType="application/json"/>
                            </response>
                        </method>
                        <resource path="/categorisations">
                            <method name="GET" id="get">
<response>
    <representation mediaType="application/json"/>
</response>
                            </method>
                        </resource>
                    </resource>
                </resource>
            </resource>
        </resource>
    </resources>
</application>
```

[1]
```json
{
  "basePath": "/rest/",
  "host": "localhost:9080",
  "paths": {
    "\\Affectation\\articles": {
      "parameters": [],
      "post": {
        "operationId": "rechercherArticle",
        "parameters": [],
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      }
    },
    "\\Affectation\\articles\\{identifiantUm}\\{siteUM}": {
      "get": {
        "operationId": "rechercherArticle",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUM",
          "required": false,
          "type": "string"
        }
      ]
    },
    "\\apgs\\{numApg: [\\p{Alnum}]{1,20}}": {
      "get": {
        "operationId": "getApg",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        }
      ]
    },
    "\\apgs\\{numApg: [\\p{Alnum}]{1,20}}\\rmes": {
      "get": {
        "operationId": "getRessourcesMetal",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        }
      ]
    },
    "\\apgs\\{numApg: [\\p{Alnum}]{1,20}}\\rmes\\{identifiantUm}\\{siteUm}": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    },
    "\\apgs\\{numApg: [\\p{Alnum}]{1,20}}\\rmes\\{identifiantUm}\\{siteUm}\\categorisations": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    },
    "\\demandesAffectation\\{reference}": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "format": "int32",
          "in": "path",
          "name": "reference",
          "required": false,
          "type": "integer"
        }
      ]
    },
    "\\rmes": {
      "get": {
        "operationId": "getRessourcesMetal",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": []
    },
    "\\rmes\\{identifiantUm}\\{siteUm}": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    },
    "\\rmes\\{identifiantUm}\\{siteUm}\\categorisations": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    }
  },
  "schemes": [
    "http"
  ],
  "swagger": "2.0"
}
```

**Comment by bobby-brennan** [comment]
[1] is the Swagger output of api-spec-converter right? Those paths look pretty wonky, looks like something went wrong

**Comment by bobby-brennan** [comment]
Can you explain how this resource path is supposed to be interpreted? 
```xml
            <resource path="/{numApg: [\p{Alnum}]{1,20}}">
```

**Comment by bobby-brennan** [comment]
I see another issue: looks like you're running on Windows, and we're using Path.join() to manipulate URL paths, which is for filesystems. We should really be using the URL package instead. Will open a separate bug

**Comment by bobby-brennan** [comment]
Pushed a couple fixes:
* Fix Path.join() calls (right now it still calls Path.join() but replaces \ with / afterward. Pretty sure this is not the best solution but it should do the trick)
* Fix up path parameters like `{numApg: [\p{Alnum}]{1,20}}` to just be `{numApg}`

Can you try running
```
npm install https://github.com/lucybot/api-spec-converter
```
And see if the problem is fixed?

Note that you can also run 
```
converted.validate()
```
to check for validation issues.

**Comment by IvanGoncharov** [comment]
@nagromc @bobby-brennan  `validate` should be supplied with callback.
```js
converted.validate(function (errors, warnings) {
});
```

**Comment by IvanGoncharov** [comment]
@nagromc @bobby-brennan Is `/{numApg: [\p{Alnum}]{1,20}}` valid path in WADL?
I read [spec](https://www.w3.org/Submission/wadl/) and couldn't find reference to anything like that.
Or I'm missing something?

**Comment by nagromc** [comment]
First, guys, thank you for your involvement.

@bobby-brennan wrote
> [1] is the Swagger output of api-spec-converter right? Those paths look pretty wonky, looks like something went wrong

Yes it is, according to the provided snippet :wink: 
```node
console.log(converted.stringify()); // outputs [1]
```

-

@bobby-brennan wrote
> Can you explain how this resource path is supposed to be interpreted?
> 
>             <resource path="/{numApg: [\p{Alnum}]{1,20}}">

The server uses Jersey v 1.19 (Java REST implementation, aka JAX-RS), and the paths can be customized with regular expressions. See [Path](http://docs.oracle.com/javaee/6/api/javax/ws/rs/Path.html#value%28%29) and [Pattern](http://docs.oracle.com/javase/1.5.0/docs/api/java/util/regex/Pattern.html#posix) for more information.

`{numApg: [\p{Alnum}]{1,20}}` means `numApg` must be an alphanumeric string between 1 and 20 characters (`\p{Alnum}` being a shortcut for `[a-zA-Z0-9]`)

-

@bobby-brennan wrote
> I see another issue: looks like you're running on Windows, and we're using Path.join() to manipulate URL paths, which is for filesystems. We should really be using the URL package instead. Will open a separate bug

Yes I am. I think that's why `api-spec-converter` outputs double backslashes `\\`. I've tried replacing `\\` with `/` but the result was the same.

-

@bobby-brennan wrote
> Can you try running
> 
>     npm install https://github.com/lucybot/api-spec-converter

I will try that and will provide the results soon. Thanks.

-

@bobby-brennan wrote
> Note that you can also run
> 
>     converted.validate()
> 
> to check for validation issues.

@IvanGoncharov wrote
> @nagromc @bobby-brennan validate should be supplied with callback.
> 
>     converted.validate(function (errors, warnings) {
>     });

I looked for a validator and I didn't know `api-spec-converter` had one. Good to know!

-

@IvanGoncharov wrote
> @nagromc @bobby-brennan Is `/{numApg: [\p{Alnum}]{1,20}}` valid path in WADL?
> I read spec and couldn't find reference to anything like that.
> Or I'm missing something?

Well, you're right. I must admit I didn't even looked at the WADL as it is automatically generated by Jersey. I will try without those regex constraints on the parameters.

**Comment by nagromc** [comment]
OK so here is the new output of `api-spec-converter` without regex constraints and with `api-spec-converter` revision [b3401e1](https://github.com/lucybot/api-spec-converter/tree/b3401e16318259e89c19ded1160437016fc49fdb):

```json
{
  "basePath": "/rest/",
  "host": "localhost:9080",
  "paths": {
    "/Affectation/articles": {
      "parameters": [],
      "post": {
        "operationId": "rechercherArticle",
        "parameters": [],
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      }
    },
    "/Affectation/articles/{identifiantUm}/{siteUM}": {
      "get": {
        "operationId": "rechercherArticle",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUM",
          "required": false,
          "type": "string"
        }
      ]
    },
    "/apgs/{numApg}": {
      "get": {
        "operationId": "getApg",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        }
      ]
    },
    "/apgs/{numApg}/rmes": {
      "get": {
        "operationId": "getRessourcesMetal",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        }
      ]
    },
    "/apgs/{numApg}/rmes/{identifiantUm}/{siteUm}": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    },
    "/apgs/{numApg}/rmes/{identifiantUm}/{siteUm}/categorisations": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "numApg",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    },
    "/demandesAffectation/{reference}": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "format": "int32",
          "in": "path",
          "name": "reference",
          "required": false,
          "type": "integer"
        }
      ]
    },
    "/rmes": {
      "get": {
        "operationId": "getRessourcesMetal",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": []
    },
    "/rmes/{identifiantUm}/{siteUm}": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    },
    "/rmes/{identifiantUm}/{siteUm}/categorisations": {
      "get": {
        "operationId": "get",
        "responses": {
          "200": {
            "description": "Successful Response"
          }
        }
      },
      "parameters": [
        {
          "in": "path",
          "name": "identifiantUm",
          "required": false,
          "type": "string"
        },
        {
          "in": "path",
          "name": "siteUm",
          "required": false,
          "type": "string"
        }
      ]
    }
  },
  "schemes": [
    "http"
  ],
  "swagger": "2.0"
}
```

It looks better, but I get an error when I had this to my Gulp task as you suggested:

```node
converted.validate(function(err, warn) {
    if (err !== null) {
        console.error('Error: WADL to Swagger2 conversion');
        console.trace(err);
    }

    if (warn !== null) {
        console.warn('Warning: WADL to Swagger2 conversion');
        console.trace(warn);
    }
});
```

The error:
```
Error: WADL to Swagger2 conversion
Trace: [ { code: 'OBJECT_MISSING_REQUIRED_PROPERTY',
    message: 'Missing required property: info',
    path: [] } ]
    at C:\xxx\gulpfile.js:67:25
    at C:\xxx\node_modules\api-spec-converter\lib\types\swagger_2.js:70:12
    at process._tickCallback (node.js:368:9)
```

Obviously, `lucy-console` crashes with the same error.

**Comment by bobby-brennan** [comment]
I just added a dummy info object to the WADL converter.  It looks like there are duplicate operationIDs still though, which I'm not sure how to handle gracefully.  E.g. in the WADL there are several operations with id=rechercherArticle

@IvanGoncharov any idea if this is legal WADL?

**Comment by bobby-brennan** [comment]
@nagromc worth noting that lucy-console will handle some invalid specs gracefully.  I just launched it with the latest output of api-spec-converter using your WADL and it works fine, even though SwaggerParser complains about duplicate operation IDs.

**Comment by IvanGoncharov** [comment]
>Well, you're right. I must admit I didn't even looked at the WADL as it is automatically generated by Jersey. I will try without those regex constraints on the parameters.

@bobby-brennan So I think it better to revert 946fa0044ec4cde3ad9ff9a74169f528d64347fd. Since we shouldn't support all non-standard features of every format.

> @IvanGoncharov any idea if this is legal WADL?

I ran validation on file and get following result:
```
test.wadl:16: element method: Schemas validity error : Element '{http://wadl.dev.java.net/2009/02}method', attribute 'id': 'get' is not a valid value of the atomic type 'xs:ID'.
test.wadl:29: element method: Schemas validity error : Element '{http://wadl.dev.java.net/2009/02}method', attribute 'id': 'rechercherArticle' is not a valid value of the atomic type 'xs:ID'.
test.wadl:41: element method: Schemas validity error : Element '{http://wadl.dev.java.net/2009/02}method', attribute 'id': 'getRessourcesMetal' is not a valid value of the atomic type 'xs:ID'.
test.wadl:45: element method: Schemas validity error : Element '{http://wadl.dev.java.net/2009/02}method', attribute 'id': 'get' is not a valid value of the atomic type 'xs:ID'.
test.wadl:49: element method: Schemas validity error : Element '{http://wadl.dev.java.net/2009/02}method', attribute 'id': 'get' is not a valid value of the atomic type 'xs:ID'.
test.wadl fails to validate
```
So it's invalid and should be fixed in WADL.

> I just added a dummy info object to the WADL converter.

I don't think it's good default behavior. You give user false sense of correctness, without even notifying him. For example his API can have version `2.0` but he will never know that you decide it for him that his API is `1.0.0`.

I think we should make function in Swagger format which fill missing required field with dummy data.
But it shouldn't be executed by default.

**Comment by nagromc** [comment]
@bobby-brennan wrote
> @nagromc worth noting that lucy-console will handle some invalid specs gracefully. I just launched it with the latest output of api-spec-converter using your WADL and it works fine, even though SwaggerParser complains about duplicate operation IDs.

I updated `api-spec-converter` to revision [54e5c83](https://github.com/lucybot/api-spec-converter/commit/54e5c836673958acc2f054fec40ee4989cf4795d), and this works with the following Gulp task:

```node
var ConsoleApp = require('express')();
var LucyConsole = require('lucy-console');
var Converter = require('api-spec-converter');

gulp.task('view-api', function() {
    Converter.convert({
        from: 'wadl',
        to: 'swagger_2',
        source: 'http://localhost:9080/rest/application.wadl'
    }, function(err, converted) {
        if (err !== null) {
            console.trace(err);
            return;
        }

        // add missing needed components
        converted.fillMissing();

        console.log(converted.stringify());

        converted.validate(function(err, warn) {
            if (err !== null) {
                console.error('Error: WADL to Swagger2 conversion');
                console.trace(err);
            }

            if (warn !== null) {
                console.warn('Warning: WADL to Swagger2 conversion');
                console.trace(warn);
            }
        });

        var apiConsole = new LucyConsole({
            swagger: converted.spec
        });

        ConsoleApp.use(apiConsole.router);
        ConsoleApp.listen(3000);
    });
});
```

The Gulp task outputs the following error as you already noticed though:
```
Error: WADL to Swagger2 conversion
Trace: [ { code: 'DUPLICATE_OPERATIONID',
    message: 'Cannot have multiple operations with the same operationId: get',
    path:
     [ 'paths',
       '/rmes/{identifiantUm}/{siteUm}/categorisations',
       'get',
       'operationId' ] },
  { code: 'DUPLICATE_OPERATIONID',
    message: 'Cannot have multiple operations with the same operationId: getRessourcesMetal',
    path: [ 'paths', '/apgs/{numApg}/rmes', 'get', 'operationId' ] },
  { code: 'DUPLICATE_OPERATIONID',
    message: 'Cannot have multiple operations with the same operationId: get',
    path:
     [ 'paths',
       '/apgs/{numApg}/rmes/{identifiantUm}/{siteUm}',
       'get',
       'operationId' ] },
  { code: 'DUPLICATE_OPERATIONID',
    message: 'Cannot have multiple operations with the same operationId: get',
    path:
     [ 'paths',
       '/apgs/{numApg}/rmes/{identifiantUm}/{siteUm}/categorisations',
       'get',
       'operationId' ] },
  { code: 'DUPLICATE_OPERATIONID',
    message: 'Cannot have multiple operations with the same operationId: get',
    path:
     [ 'paths',
       '/demandesAffectation/{reference}',
       'get',
       'operationId' ] },
  { code: 'DUPLICATE_OPERATIONID',
    message: 'Cannot have multiple operations with the same operationId: rechercherArticle',
    path: [ 'paths', '/Affectation/articles', 'post', 'operationId' ] } ]
    at L:\mcourbet\dev\git\plq\plq-parent\plq-web\gulpfile.js:69:25
    at L:\mcourbet\dev\git\plq\plq-parent\plq-web\node_modules\api-spec-converter\lib\types\swagger_2.js:81:12
    at process._tickCallback (node.js:368:9)
```

I need to find a way to modify the WADL to make it compliant.

**Comment by nagromc** [comment]
FYI, I put back

<pre>
/{numApg<b>: [\p{Alnum}]{1,20}</b>}
</pre>

and it doesn't seem to be a problem.

`api-spec-converter` strips away the regex constraint:

```json
"/apgs/{numApg}": {
  "get": {
    "operationId": "getApg",
    "responses": {
      "200": {
        "description": "Successful Response"
      }
    }
  },
  "parameters": [
    {
      "in": "path",
      "name": "numApg",
      "required": true,
      "type": "string"
    }
  ]
},
```

In `lucy-console`:
![Imgur](http://i.imgur.com/4rSRo1J.png)

That's okay for now. I don't know well WADL and Swagger, but is it possible to add this kind of regex constraint in a "compliant way"?

**Comment by bobby-brennan** [comment]
It's possible to add regex and other validators to Swagger, but it could be
difficult to convert from Jersey format.

If you're familiar with how these validators are constructed in Jersey,
feel free to take a stab at it and send over a PR.  You can look at the
Swagger definition for parameters to see the different validation fields:
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameterObject

Bobby Brennan
bobby.brennan@gmail.com

**Comment by IvanGoncharov** [comment]
@nagromc We released version `1.9.0`.
If you find other issues please report it here.
We really need user input to improve our conversion tool.
